### PR TITLE
Feature requests

### LINKED ISSUE #32: Model and Molecule Deletion
Implement ability to delete saved models & molecules

### LINKED ISSUE #37: Add learning rate as a customizable parameter

### PR DESCRIPTION
**Pull request by CSCMe**
Depends on #36 

Changelog:
- Fixed most tests
- Added Molecule Renaming/Overwriting when name or smiles code already used by another molecule
- Added learning rate as a parameter to trainings. Closes #37 
- Added Ability to delete Models, Molecules & individual fittings. Closes #32 
- Added learning rate to scoreboard

Issues:
- [x] Inconsistencies on molecule deletion (different behaviour last molecule deleted, some molecule deleted)

### CODE REVIEW

File: backend/machine_learning/ml_functions.py [hunk 1]
```diff
@@ -89,21 +92,22 @@ def stop_training(self):
...
+        new_training = Training(user_id, dataset_id, model_id, labels, epochs, learning_rate, batch_size, fitting_id)
         with lock:
             live_trainings[user_id] = new_training
         new_training.start_training()
 
-    except (TypeError, AttributeError, ValueError):
+    except (TypeError, AttributeError, ValueError) as e:
         with lock:
             del live_trainings[user_id]
+        print(e)
```
Comment (13thWitch) [inline backend/machine_learning/ml_functions.py:110]: Is this error printout intentionally still there?

**Review by 13thWitch — APPROVED** [review]
This is great. Ready to merge.
